### PR TITLE
Import refacto

### PR DESCRIPTION
**Pull request by braktar**

### CODE REVIEW

File: app/jobs/importer_base.rb [hunk 1]
```diff
@@ -31,59 +31,64 @@ def initialize(customer)
...
-
-          dests = data.each_with_index.collect{ |row, line|
-            # Switch from locale or custom to internal column name in case of csv
-            row = yield(row, line + 1 + (options[:line_shift] || 0))
+      Destination.without_callback(:validation, :before, :update_geocode) do
+        VehicleUsageSet.without_callback(:create, :before, :check_max_vehicle_usage_set) do
+          Customer.transaction do
+            before_import(name, data, options)
+
+            dests = data.map.with_index{ |row, line|
```
Comment (frodrigo) [inline app/jobs/importer_base.rb:51]:
```suggestion
            dests = data.with_index{ |row, line|
```

File: app/jobs/importer_base.rb [hunk 2]
```diff
@@ -31,59 +31,64 @@ def initialize(customer)
...
+              # Switch from locale or custom to internal column name in case of csv
+              row = yield(row, line + 1 + (options[:line_shift] || 0))
+
+              next if row.empty? # Skip empty line
+
+              begin
+                if (ref = uniq_ref(row))
+                  if refs.key?(ref)
```
Comment (frodrigo) [inline app/jobs/importer_base.rb:59]: If ref is an Array it will not work.

Comment (braktar) [reply app/jobs/importer_base.rb:59]: That was one of my interrogations. Nevertheless, an Array can be key of a Hash.

File: app/jobs/importer_base.rb (hunk 1 above)
Comment (braktar) [reply app/jobs/importer_base.rb:51]: data might be a Hash or an Array, `with_index` is not defined not defined on theses

**Pull request merged**